superset/assets/src/utils/common.js

```diff
@@ -108,3 +108,18 @@ export function prepareCopyToClipboardTabularData(data) {
   }
   return result;
 }
+
+
+// More functional switch cases
+// https://codeburst.io/alternative-to-javascripts-switch-statement-with-a-functional-twist-3f572787ba1c
+export const matched = x => ({
+  on: () => matched(x),
+  otherwise: () => x,
+});
+
+export function match(x) {
```

Comment:
- Can make this `utils/match.js` instead of adding to `common.js`.
- I usually write functional code when applicable, but not sure if we have to use purely functional code for this though. It seems like the code above that use `match()` would be fine using `switch`. May be even more straightforward that this arbitrary implementation of `match`.